lib/bsnutil/util.go

```diff
@@ -0,0 +1,202 @@
+package bsnutil
+
+import (
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/binary"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+const (
+	tokenPrefix        = "token-"
+	pseudonymPrefix    = "ps-"
+	minTokenLength     = len(tokenPrefix) + 1     // "token-" + at least 1 char
+	minPseudonymLength = len(pseudonymPrefix) + 1 // "ps-" + at least 1 char
+
+	// Input validation limits
+	maxAudienceLength = 255
+	minBSN            = 100000000
+	maxBSN            = 999999999
+)
+
+// CreateTransportToken creates a transport token from BSN and audience using simple XOR transformation.
+// Each transport token is unique (includes random nonce) but contains the same encrypted BSN.
+// This ensures transport tokens cannot be tracked while the NVI can always generate the same pseudonym.
+//
+// Parameters:
+//   - bsn: The Dutch social security number (BSN)
+//   - audience: The identifier for the organization/audience receiving the token
+//
+// Returns a transport token in format: "token-{audience}-{transformedBSN}-{nonce}"
+func CreateTransportToken(bsn int, audience string) (string, error) {
```

Comment:
I'd make BSN a `string`, as it fixed length, at some systems may use prefix-zeroes